roles/kafka_broker/tasks/set_principal.yml

```diff
@@ -2,22 +2,22 @@
 - name: Set Principal - Sasl Scram
   set_fact:
     kafka_broker_principal: "User:{{ sasl_scram_users_final.admin.principal }}"
-  when: listener['sasl_protocol'] | default(sasl_protocol) | confluent.platform.normalize_sasl_protocol == 'SCRAM-SHA-512'
+  when: "'SCRAM-SHA-512' in (listener['sasl_protocol'] | default(sasl_protocol) | confluent.platform.normalize_sasl_protocol)"
 
 - name: Set Principal - Sasl Scram 256
   set_fact:
     kafka_broker_principal: "User:{{ sasl_scram256_users_final.admin.principal }}"
```

Comment:
Since we can have multiple protocols on a single listener so should we be adding principal corresponding to each principal in the kafka_broker_principals list instead of overriding principals to only have the one which comes last in code flow ?

Reply:
Fixed the code to consider only the first value for setting the principal after discussion with @mansisinha

Reply:
I didnt get it. Why just add the first and not all ?

Reply:
@rrbadiani  we can only have one mechanism for `sasl.mechanism.inter.broker.protocol`, only the corresponding principals are needed here